client/src/main/java/io/split/client/SplitClientImpl.java

```diff
@@ -234,6 +240,17 @@ public boolean track(String key, String trafficType, String eventType, double va
         return track(event);
     }
 
+    @Override
+    public void blockUntilReady() throws TimeoutException, InterruptedException {
```

Comment:
```java
if (_config.blockUntilReady() <= 0) {
    throw new IllegalArgumentException("setBlockUntilReadyTimeout must be positive but in config was: " + _config.blockUntilReady());
}
if (!_gates.isSDKReady(_config.blockUntilReady())) {
    throw new TimeoutException("SDK was not ready in " + _config.blockUntilReady()+ " milliseconds");
 }
```

Reply:
see if it can be written that way to simplicity.

Reply:
done